packages/flutter/lib/src/material/tabs.dart

```diff
@@ -387,81 +388,104 @@ abstract class TabBarSelectionPerformanceListener {
   void handleSelectionDeactivate();
 }
 
-class TabBarSelection extends StatefulComponent {
+class TabBarSelection<T> extends StatefulComponent {
   TabBarSelection({
     Key key,
-    this.index,
-    this.maxIndex,
+    this.value,
+    Iterable<T> values,
     this.onChanged,
     this.child
-  }) : super(key: key)  {
+  }) : _values = new List.unmodifiable(values), super(key: key)  {
```

Comment:
I'm happy to replace the defensive copy with a a comment/warning.

Good point about checking for duplicates.